go/client/cmd_wallet_cancel_all.go

```diff
@@ -0,0 +1,196 @@
+	"golang.org/x/net/context"
+	"golang.org/x/sync/semaphore"
+)
+
+type cmdWalletCancelAll struct {
+	libkb.Contextified
+	infile  string
+	outfile string
+}
+
+func newCmdWalletCancelAll(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	cmd := &cmdWalletCancelAll{
+		Contextified: libkb.NewContextified(g),
+	}
+	return cli.Command{
+		Name:     "cancel-all",
+		Aliases:  []string{"reclaim-all"},
+		Usage:    "Cancel all pending payments",
+		Unlisted: true,
+		Action: func(c *cli.Context) {
+			cl.ChooseCommand(cmd, "cancel-all", c)
+		},
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "i, infile",
+				Usage: "Specify an input file.",
+			},
+			cli.StringFlag{
+				Name:  "o, outfile",
+				Usage: "Specify an output file (none by default).",
+			},
+		},
+	}
+}
+
+func (c *cmdWalletCancelAll) ParseArgv(ctx *cli.Context) (err error) {
+	if len(ctx.Args()) > 0 {
+		return UnexpectedArgsError("wallet cancel-all")
+	}
+	c.outfile = ctx.String("outfile")
+	c.infile = ctx.String("infile")
+	return nil
+}
+
+func (c *cmdWalletCancelAll) getInTxIDs(out map[string]bool) (in []string, err error) {
+	if len(c.infile) == 0 {
+		return in, nil
+	}
+
+	file, err := os.Open(c.infile)
+	if err != nil {
+		return in, err
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		txid := scanner.Text()
+		if _, ok := out[txid]; !ok {
+			in = append(in, txid)
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return in, err
+	}
+	return in, nil
+}
+
+func (c *cmdWalletCancelAll) getOutTxIDs() (out map[string]bool, err error) {
+	out = make(map[string]bool)
+	if len(c.outfile) == 0 {
+		return out, nil
+	}
+
+	file, err := os.Open(c.outfile)
+	if err != nil {
+		return out, err
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		row := scanner.Text()
+		out[row] = true
+	}
+	if err := scanner.Err(); err != nil {
+		return out, err
+	}
+	return out, nil
+}
+
+func (c *cmdWalletCancelAll) Run() (err error) {
+	defer transformStellarCLIError(&err)
+	dui := c.G().UI.GetDumbOutputUI()
+	cli, err := GetWalletClient(c.G())
+	if err != nil {
+		return err
+	}
+
+	out, err := c.getOutTxIDs()
+	if err != nil {
+		return err
+	}
+	in, err := c.getInTxIDs(out)
+	if err != nil {
+		return err
+	}
+
+	var outfile *os.File = nil
+	// if c.outfile defined, append finished txIDs
+	if len(c.outfile) > 0 {
+		outfile, err = os.OpenFile("out.txt",
+			os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer outfile.Close()
+	} else {
+		dui.Printf("Note: no outfile specified, so not keeping track of complete txIDs.\n")
+	}
+
+	maxWorkers := 100 // OR runtime.GOMAXPROCS(0)
+	sem := semaphore.NewWeighted(int64(maxWorkers))
+	dui.Printf("Canceling %d txIDs with %d maxWorkers; %d txIDs already complete...\n", len(in), maxWorkers, len(out))
+
+	var wg sync.WaitGroup
+	var successCount, skipCount, failCount, writeErrorCount int
+	for _, txID := range in {
```

Comment:
could you trap SIGTERM, SIGINT to cleanly exit this loop (and thus gracefully shut down this command)?

Reply:
done